### PR TITLE
Actually hook up DnaCompatParams

### PR DESCRIPTION
**Pull request by maackle**
### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_integrity_types/CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,20 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+## 0.3.0-beta-dev.24
+
+## 0.3.0-beta-dev.23
+
+## 0.3.0-beta-dev.22
+
+## 0.3.0-beta-dev.21
+
+## 0.3.0-beta-dev.20
+
```
Comment (maackle) [inline crates/holochain_integrity_types/CHANGELOG.md:39]:
```suggestion
```

File: crates/holochain/src/conductor/conductor.rs [hunk 2]
```diff
@@ -1338,23 +1334,52 @@ mod app_impls {
...
-                agent_key,
-                data,
-            )
-            .await;
+        ) -> ConductorResult<Vec<DnaHash>> {
+            use crate::sweettest::DnaWithRole;
 
-            self.install_app_bundle(payload).await?;
+            let dnas_with_roles: Vec<_> = data.iter().map(|(dr, _)| dr).collect();
+            let manifest = crate::sweettest::app_manifest_from_dnas(&dnas_with_roles);
```
Comment (neonphog) [inline crates/holochain/src/conductor/conductor.rs:1350]: time to move this out of sweettest then?

File: crates/holochain/tests/test_utils/mod.rs [hunk 3]
```diff
@@ -264,6 +263,8 @@ pub async fn register_and_install_dna_named(
 
     let resources = vec![(dna_path.clone(), dna_bundle)];
 
+    dbg!(&manifest);
+
```
Comment (neonphog) [inline crates/holochain/tests/test_utils/mod.rs:267]:
```suggestion
```

File: crates/holochain/src/sweettest/sweet_conductor.rs [hunk 4]
```diff
@@ -347,19 +358,21 @@ impl SweetConductor {
...
         let dnas_with_proof: Vec<_> = dnas_with_roles
             .iter()
             .map(|dr| (dr.to_owned(), None))
             .collect();
-        self.raw_handle()
+        let dna_hashes = self
+            .raw_handle()
             .install_app_minimal(installed_app_id.clone(), agent, &dnas_with_proof)
             .await?;
+        // app.all_cells()
```
Comment (neonphog) [inline crates/holochain/src/sweettest/sweet_conductor.rs:372]:
```suggestion
```

**Review by neonphog — APPROVED** [review]

File: crates/holochain_zome_types/src/dna_compat.rs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+//! DNA compatibility parameters
+
+use holo_hash::DnaHashB64;
+use holochain_serialized_bytes::prelude::*;
+
+/// Extra parameters that contribute to determining the DNA hash.
+/// They are set by the conductor at install time and cannot be specified
+/// by the DNA developer.
+/// They represent different aspects of networking compability.
```
Comment (jost-s) [inline crates/holochain_zome_types/src/dna_compat.rs:9]:
```suggestion
/// They represent different aspects of networking compatibility.
```

File: crates/holochain_zome_types/src/dna_def.rs [hunk 6]
```diff
@@ -43,6 +43,12 @@ pub struct DnaDef {
     /// computation.
     pub modifiers: DnaModifiers,
 
+    /// Parameters defined by the conductor into which this DNA is installed.
+    /// These specify various constraints on network compatibility based on the
+    /// runtime environment. These do affect the DNA hash.
```
Comment (jost-s) [inline crates/holochain_zome_types/src/dna_def.rs:48]:
```suggestion
    /// runtime environment and do affect the DNA hash.
```

**Review by jost-s — APPROVED** [review]
In light of the office hours discussion, do you consider another name for the params? Perhaps network compat or conductor compat?

File: crates/holochain_zome_types/src/dna_compat.rs (hunk 5 above)
Comment (maackle) [reply crates/holochain_zome_types/src/dna_compat.rs:9]: In light of the office hours, I'm probably going to scrap this PR altogether!

Comment (jost-s) [reply crates/holochain_zome_types/src/dna_compat.rs:9]: I was gonna say that at first actually.

**Comment by maackle** [comment]
We decided that this is the wrong direction. The new direction is #3415

**Pull request closed without merging**